lib/wallet.js

```diff
@@ -695,6 +709,80 @@ Wallet.prototype.upgradeKeyIndex = function(keyIndex, cb) {
     return deferred.promise;
 };
 
+/**
+ * generate a new derived private key and return the rootstock address for it
+ *
+ * @param [chainIdx] int
+ * @param [cb]  function        callback(err, address)
+ * @returns {q.Promise}
+ */
+Wallet.prototype.getRskAddress = function(chainIdx, cb) {
+    var self = this;
+
+    // chainIdx is optional
+    if (typeof chainIdx === "function") {
+        cb = chainIdx;
+        chainIdx = null;
+    }
+
+    var deferred = q.defer();
+    deferred.promise.spreadNodeify(cb);
+
+    // Only enter if it's not an integer
+    if (chainIdx !== parseInt(chainIdx, 10)) {
+        // deal with undefined or null, assume defaults
+        if (typeof chainIdx === "undefined" || chainIdx === null) {
+            chainIdx = Wallet.CHAIN_RSK_DEFAULT;
+        } else {
+            // was a variable but not integer
+            deferred.reject(new Error("Invalid chain index"));
+            return deferred.promise;
+        }
+    }
+
+    // Get Rootstock address from server.
+    deferred.resolve(self.sdk.getNewDerivation(self.identifier, "M/" + self.keyIndex + "'/" + chainIdx)
+        .then(function(newDerivation) {
+            var path = newDerivation.path.replace("\'","").replace('9999',"9999'");
```

Comment:
not sure this behaves as intended - it seems to remove all `'`s, placing in 9999' wherever 9999 is encountered, which can also be found in the level we use for address derivations

what's it supposed to do? is it necessary?

Reply:
Right, removing this as the path is parsed later on in Wallet.deriveByPath. This is a relic from testing.